flume-ng-doc/sphinx/FlumeUserGuide.rst

```diff
@@ -234,6 +234,23 @@ The original Flume terminal will output the event in a log message.
 
 Congratulations - you've successfully configured and deployed a Flume agent! Subsequent sections cover agent configuration in much more detail.
 
+Using environment variables in configuration files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If we enable environment variable interpolation by adding `-DpropertiesImplementation=org.apache.flume.node.EnvVarResolverProperties`
+to our `bin/flume-ng` call, we can use environment variables in the config, such as::
+
+  a1.sources = r1
+  a1.sources.r1.type = netcat
+  a1.sources.r1.bind = 0.0.0.0
+  a1.sources.r1.port = ${NC_PORT}
+  a1.sources.r1.channels = c1
+
+So a complete command line start could look like::
+
+  $ NC_PORT=44444 bin/flume-ng agent --conf conf --conf-file example.conf --name a1 -Dflume.root.logger=INFO,console -DpropertiesImplementation=org.apache.flume.node.EnvVarResolverProperties
+
+
```

Comment:
I think we should be explicit about a couple of different things in the user guide

1) This only affects values and not keys
2) You can use env variables however you want, the example is just an example. 

I would also review the language a little bit (eg, remove "we")